### PR TITLE
[Estuary] rework DialogMusicInfo.xml

### PR DESCRIPTION
**Pull request by DaVukovic**
## Description

This change makes the musicinfo dialog a bit more useful as it shows some more of the infolabels. 

Changes:
- changed the general layout
- added a new box to which one is also able to navigate to
- squashed the buttons at the album info to show them all
- also tried to take care about the animations and which part of the section slide in from which side. Hope I haven't missed anything. 

Credits also going to @HitcherUK where I stole nearly the complete infolabel section from the Confluence skin. Hope this is fine.

I guess there might be more improvements possible for that section and I'm all ears for any suggestion. As this is an improvement I guess it might not be backported to Kodi 18. So we have plenty of time to get this done if any changes are required.

## Motivation and Context

The musicinfo dialog was not as informative as it could be. So @DaveTBlake asked for some help to get more infos in it, and here it is ;)

## How Has This Been Tested?

I tried this on current Kodi master with a sample MyMusicXY.db from Dave and also with my own library where the screenshots below are coming from. I also tested different tags for showing which instruments are used by whom (songinfo screen).

## Screenshots (if appropriate):

Artist info:
music1

Album info:
music2

Song info:
music3

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [x] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Review by Hitcher — COMMENTED** [review]
Looks good to me.

**Comment by msduketown** [comment]
Jah man

**Review by ronie — COMMENTED** [review]
nice! i've always been in favour of displaying all the info we have for an item.

two minors to fix:
**album info**: there's not enough room for the buttons at the bottom, they slide a few pix to the left when you navigate to the last button.
**song info**: i can't focus the info list (song has no comment, so fanart is shown)
likely the same issue exists in the artist/album info dialogs when description/review is empty.

**Comment by DaVukovic** [comment]
Thanks for the hint ;)

Done the changes accordingly. I missed the part that there are 2 buttons with id "130" and also changed the itemgap for the bottom buttons.

**Comment by ksooo** [comment]
If you ask me, putting even more stuff in one dialog is not always good. Some of the new dialogs look rather crowded to me. Just my 2 ct.

**Comment by ronie** [comment]
imho it's the purpose of these info dialogs...
you can keep the regular library views clean and simple, and have one place where all detailed info of an item is shown.

**Review by DaveTBlake — APPROVED** [review]
Thanks for this @DaVukovic, the original music information dialog was pretty useless, this now actually displays the information about the artist, album or song.

**Comment by DaveTBlake** [comment]
>imho it's the purpose of these info dialogs... you can keep the regular library views clean and simple, and have one place where all detailed info of an item is shown.

Yes absolutely agree.

**Comment by da-anda** [comment]
nothing against displaying all the info we know in those dialogs, but I'm with ksooo that these are way too crowded now with too little whitespace between certain elements. Also the `song info` dialog makes absolutely no sense as it is now, since information is redundant. The entire top area right of the album cover is basically the same info as in the right column, so IMO all this info should at least move to this column.

**Comment by DaveTBlake** [comment]
>The entire top area right of the album cover is basically the same info as in the right column

That is not correct @da-anda although there is some repetition. There are also limitatioins with the way the skin can access the data. The various musicians and other way people contribute to the song is provided as a list so shown in a container (I think that is the right term) right of cover. The other information (shown right column) is via info labels. Some information e.g. conposer is available via both list and infolabel

**Comment by ksooo** [comment]
I'm not a designer, but the old layout seems very professional and eye catching to me. Clear structure, symmetric, etc. The new layout is a bit like stuff dropped here and there, missing the esthetics the old layout had.

**Comment by da-anda** [comment]
> That is not correct @da-anda although there is some repetition. There are also limitatioins with the way the skin can access the data. The various musicians and other way people contribute to the song is provided as a list so shown in a container (I think that is the right term) right of cover. The other information (shown right column) is via info labels. Some information e.g. composer is available via both list and infolabel

well, fact is, that apart from the genre and the file path, all the info was already present in that dialog and is now redundant (while cluttering the dialog). I'm not saying that we should not improve/enrich those dialogs and welcome a change here, but we might have to rethink those dialogs in their entirety, along with the video info ones (and likely others like PVR, ...), so that there is a consistency in the design. Estuary uses certain spacing for certain things, and this is also lost here.

@DaVukovic if you like, I could try to do some mockups

Nevertheless - feel free to ignore my 2cent

**Comment by DaveTBlake** [comment]
>we might have to rethink those dialogs in their entirety

That is fine, but if that does happen then can the designers please ensure that they have a fully populated music library to test their creations. Also take advance warning that even more song information will be available sometime in v19 (dates, BPM etc.), so allow for expansion.

The purpose of the information dialogs is not to be decorative but to be informative. Personally I find having information scattered about the screen far harder to take in and assimilate than when it is in a simple list of  properties.  Just looking at a screen you may notice how eye-catching it is, but you also need to use a UI purposefully to fully evaluate any design.  Hopefully both elegance and functionality can be achieved. 

Meanwhile, awaiting an entire Estuary dialog layout redesign, could the data that has been missing since release, especially the artist and album information, please be added (as this PR does).  At very least it is a clear reminder that there is this data to be shown on any new future layout, rather than getting totally forgotten again.

**Comment by da-anda** [comment]
@DaveTBlake I think technically it might make sense to provide the aforementioned `list` (with writer, composer, ...) for any fileitem (pvr, video, music, game) and extend it with new properties as they get added to the core. That way skins could use this list to display "all item properties/metadata" in a compact table and display/pick certain properties on selected places via infolabels. Would that sound good? If so, would you mind having a look if and how we could unify this?

**Comment by zag2me** [comment]
Excellent improvement. Mirrors what many other music systems show on their info screens.

**Comment by ksooo** [comment]
I'm in no way against the new information. My point is that visual appearance possibly should be improved and realigned with Estuary general design guidelines.

**Comment by DaVukovic** [comment]
Thanks for all the comments.

@ksooo could you please give me a hint what you mean exactly by "Estuary desing guidelines"? It's not that I'm against any kind of change. This is not written into stone. But I'm not aware of any kind of guidelines. 
Generally I thought that 
- changing the size of a cover
- adding a new box for infolabels
- changing the width of the description box

is not against anything. You can't compare that with the TV show info, because DVD/BR covers are rectangular whilst CD album covers are quadratic. And I had to decide which part has to be changed to get more space to use. 

But as said, I'm open for any suggestion.

**Comment by ksooo** [comment]
> "Estuary design guidelines"

Not sure there are (written) guidelines. 

Example 1: From what I observed, all dialogs having a row of buttons at the bottom seem to put those buttons at the same distance from bottom of the screen. You moved the button rows down to the very bottom of the screen, which breaks which seems to be a design pattern across all Estuary dialogs.

Example 2: Not special for Estuary, though. Controls are not "properly aligned". Look at the song text boxes. In the original design there are two of them in a row, most left box starts exactly where the textbox with larger descriptive text below the song boxes starts and most right song box ends where  the textbox with larger descriptive text ends. In your design, most right of second song box is somewhere in the middle of the description box and one screenshot shows even half of a third song box. This simply does not look Good (TM).

**Comment by DaVukovic** [comment]
Would we agree more if I would make it compliant to the movie info screen?

For example:
example

It will be less space for everything. But at least it would be compliant to something we already have. Which generally is already not the case on current Estuary as the buttons from the actual music info screen aren't on the same height as those from the movie or tv show info screen. 

I'm only asking as I'm really willing to change that but I would like to know to what we might agree on.

**Comment by ronie** [comment]
yeah, that might be the best idea.
perhaps move the thumb up and fill the empty gap with a small fanart imge

**Comment by ksooo** [comment]
yeah, I like that.

**Comment by da-anda** [comment]
I would not center the cover in the available space but align it on the top edge. On the space below you could display the most relevant info labels or something that is visually "light weight" (larger font with decent spacing between lines). I would not put the fanart there since it will again clutter the view and leave no place for the eyes to rest.
As mentioned earlier - I can try to make some mockups, but those will take time. I likely won't be able to work on those before Devcon.

**Comment by DaVukovic** [comment]
I'm feeling sorry that this took a bit longer than I expected. Ran a bit out of time.

First "forced push" was the rebase, 2nd one is the actual change

I made it compliant to the tv-show info screen as best as I could:

Artist info:
grafik

Album info. The buttons on the bottom don't move anymore:
grafik

Song info: 
grafik

**Comment by DaveTBlake** [comment]
Thanks for your efforts @DaVukovic , I leave the others to comment on aesthetic standards etc.

**Comment by da-anda** [comment]
much better IMO. I'd align the cover at the top edge though and not center it

**Review by ronie — CHANGES_REQUESTED** [review]
two issues i've noticed:

1 in the song info dialog, you can't navigate from the fanart image to the list with infolabels

songinfo

2 you're using the wrong focused texture in the list with infolabels
(see comparison between music & video infodialogs)

focus

File: addons/skin.estuary/xml/DialogMusicInfo.xml [hunk 1]
```diff
@@ -414,11 +445,105 @@
...
+									<label>$INFO[ListItem.Label2]</label>
+								</control>
+							</itemlayout>
+							<focusedlayout height="35">
+								<control type="image">
+									<left>0</left>
+									<top>1</top>
+									<width>430</width>
+									<height>33</height>
+									<texture border="8" colordiffuse="button_focus">buttons/thumbnail_focused.png</texture>
```
Comment (ronie) [inline addons/skin.estuary/xml/DialogMusicInfo.xml]: that's the incorrect texture i was referring to.
please use `<texture colordiffuse="button_focus">lists/focus.png</texture>` instead

Comment (DaVukovic) [reply addons/skin.estuary/xml/DialogMusicInfo.xml]: I swear I changed that to the texture you mentioned. Must be to that GH issue we talked about. Will chec if there are other flaws before pushing the next changes

File: addons/skin.estuary/xml/DialogMusicInfo.xml [hunk 2]
```diff
@@ -414,11 +445,105 @@
...
+									<label>$INFO[ListItem.Label2]</label>
+								</control>
+							</itemlayout>
+							<focusedlayout height="35">
+								<control type="image">
+									<left>0</left>
+									<top>1</top>
+									<width>430</width>
+									<height>33</height>
+									<texture border="8" colordiffuse="button_focus">lists/focus.png</texture>
```
Comment (ronie) [inline addons/skin.estuary/xml/DialogMusicInfo.xml]: the `border` attribute is not needed with the new texture, could you please drop it?

Comment (DaVukovic) [reply addons/skin.estuary/xml/DialogMusicInfo.xml]: done

**Comment by ronie** [comment]
thx!

**Comment by Rechi** [comment]
@ronie please always set the milestone before or right after merging a PR.

**Comment by ronie** [comment]
thx for the reminder, done

**Pull request merged**